### PR TITLE
Improves javascript structure

### PR DESCRIPTION
**Pull request by captn3m0**
- Uses wp_enqueue_script and wp_localize_script

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    var successMsg = document.getElementById('msg-razorpay-success');
+    successMsg.style.display = 'block';
+    document.getElementById('razorpay_payment_id').value =
+      payment.razorpay_payment_id;
+    document.getElementById('razorpay_signature').value =
+      payment.razorpay_signature;
+    document.razorpayform.submit();
+  };
+
+  console.log(data);
```
Comment (mayankamencherla) [inline script.js]: @captn3m0 Is this needed?

File: razorpay-payments.php [hunk 2]
```diff
@@ -361,16 +355,28 @@ function getOrderCreationData($orderId)
...
          **/
-        function generateOrderForm($redirectUrl, $json)
+        function generateOrderForm($redirectUrl, $data)
         {
-            $html = <<<EOT
-<script src="https://checkout.razorpay.com/v1/checkout.js"></script>
-<script>
-    var data = $json;
-</script>
+            wp_enqueue_script('razorpay_wc_script', plugin_dir_url(__FILE__)  . 'script.js',
```
Comment (mayankamencherla) [inline razorpay-payments.php]: @captn3m0 Shouldn't we register the script first and then enqueue it? 
http://www.wpbeginner.com/wp-tutorials/how-to-properly-add-javascripts-and-styles-in-wordpress/

File: razorpay-payments.php [hunk 3]
```diff
@@ -361,16 +355,28 @@ function getOrderCreationData($orderId)
...
+            wp_localize_script('razorpay_wc_script',
+                'razorpay_wc_checkout_vars',
+                $data
+            );
+
+            wp_register_script('razorpay_checkout',
+                'https://checkout.razorpay.com/v1/checkout.js',
+                null, null);
+
+            wp_enqueue_script('razorpay_wc_script');
```
Comment (mayankamencherla) [inline razorpay-payments.php]: @captn3m0 Why enqueue the script twice?

File: razorpay-payments.php [hunk 4]
```diff
@@ -361,16 +355,28 @@ function getOrderCreationData($orderId)
...
-</script>
+            wp_enqueue_script('razorpay_wc_script', plugin_dir_url(__FILE__)  . 'script.js',
+                array('razorpay_checkout'));
+
+            wp_localize_script('razorpay_wc_script',
+                'razorpay_wc_checkout_vars',
+                $data
+            );
+
+            wp_register_script('razorpay_checkout',
```
Comment (mayankamencherla) [inline razorpay-payments.php]: Shouldn't we register this script before `razorpay_wc_script` in order for the steps to be in the right logical order, even though they don't necessarily have to be in that order for it to work?

File: razorpay-payments.php [hunk 5]
```diff
@@ -361,16 +355,28 @@ function getOrderCreationData($orderId)
...
+                null, null);
+
+            wp_enqueue_script('razorpay_wc_script');
+
+            return <<<EOT
 <form name='razorpayform' action="$redirectUrl" method="POST">
     <input type="hidden" name="razorpay_payment_id" id="razorpay_payment_id">
     <input type="hidden" name="razorpay_signature"  id="razorpay_signature" >
+    <!-- This distinguishes all our various wordpress plugins -->
+    <input type="hidden" name="razorpay_wc_form_submit" value="1">
```
Comment (mayankamencherla) [inline razorpay-payments.php]: Nice.

File: script.js (hunk 1 above)
Comment (captn3m0) [inline script.js]: will remove, my bad

File: razorpay-payments.php (hunk 2 above)
Comment (captn3m0) [inline razorpay-payments.php]: I copied this from gf plugin, I think. will check and fix

**Pull request merged**